### ISSUE TITLE
longtable 4.16 : dimension too large when inside vbox ; works in 4.13

### ISSUE THREAD

**Issue by ysalmon** [latex3/latex2e#670]
## Brief outline of the bug
longtable causes a Dimension too large error when it is typeset in a vbox

## Minimal example showing the bug
(firsthead and head are identical due to example minimisation ; same for foot and lastfoot)
```latex
\RequirePackage{latexbug}
\documentclass{article}
\usepackage{longtable}
\begin{document}
\vbox{\begin{longtable}{|l|l|l|l|l|l|}
\hline
\textbf{classe}&\textbf{Copies}&\textbf{Note min}&\textbf{Note max}&\textbf{Moyenne}&\textbf{ecart-type}\\\hline\endfirsthead
\hline
\textbf{classe}&\textbf{Copies}&\textbf{Note min}&\textbf{Note max}&\textbf{Moyenne}&\textbf{ecart-type}\\\hline\endhead
\hline\endfoot
\endlastfoot
MPSI1&16&2.40&16.03&7.75&3.78\\\hline
\end{longtable}}

\end{document}
```
Removing the vbox around makes it compile. No problem with the 4.13 version of longtable.

## Log file (required) and possibly PDF file  

[fiches_juin.log](https://github.com/latex3/latex2e/files/7188100/fiches_juin.log)

**Comment by davidcarlisle** [comment]
Thanks for the file. the newer code checks the amount on the page in a better way to avoid over-runs when there are floats, or shrinkable glue on the page, but `\pagegoal` is `\maxdimen` inside the box  as the box hasn't yet been added to the page and the calculation overflows (and would  do the wrong thing anyway)

I need to detect this case and avoid the error, but will need to think where to do that.

```
\vbox{\dimendef\pagegoal=0 \pagegoal=0pt
```

will get your document working (but that is not a fix, just a quick workaround)

**Comment by ysalmon** [comment]
Thank you so much for your quick help.
Your addition makes it work under 4.18 but breaks under 4.13 ; do you know how I could make a version that works on both versions (not my colleagues will probably have updated to 4.18 yet) ?

**Comment by ysalmon** [comment]
Thank you so much for this quick workaround !

**Comment by josephwright** [comment]
Is this one that needs v5?

**Comment by rdicosmo** [comment]
## Motivation for reopening
I would kindly like to reopen this issue, as it is still unsolved, and it turns out that now the bug prevents using longtable inside a beamer block.

## Minimal example reused in beamer

```
\documentclass{beamer}
\usepackage{longtable}
\begin{document}
\begin{frame}
    \begin{longtable}{|l|l|l|l|l|l|}
      \hline
      \textbf{classe}&\textbf{Copies}&\textbf{Note min}&\textbf{Note max}&\textbf{Moyenne}&\textbf{ecart-type}\\\hline\endfirsthead
      \hline
      \textbf{classe}&\textbf{Copies}&\textbf{Note min}&\textbf{Note max}&\textbf{Moyenne}&\textbf{ecart-type}\\\hline\endhead
      \hline\endfoot
      \endlastfoot
      MPSI1&16&2.40&16.03&7.75&3.78\\\hline
    \end{longtable}
\end{frame}
\end{document}
```
## Log file

[bugreport.log](https://github.com/latex3/latex2e/files/13967922/bugreport.log)

**Comment by josephwright** [comment]
@rdicosmo  A `longtable` simply doesn't make sense in a `beamer` slide

**Comment by rdicosmo** [comment]
I agree :-) But this worked until 4.13, and it was practical sometimes to paste into slides material from an article.
Getting this strange new error when upgrading LaTeX is puzzling and makes one loose quite a bit of time to chase down what's happening :-(
